cmd/iceberg/main.go

```diff
@@ -0,0 +1,332 @@
+// under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/apache/iceberg-go"
+	"github.com/apache/iceberg-go/catalog"
+	"github.com/apache/iceberg-go/table"
+	"github.com/docopt/docopt-go"
+)
+
+const usage = `iceberg.
+
+Usage:
+  iceberg list [options] [PARENT]
+  iceberg describe [options] [namespace | table] IDENTIFIER
+  iceberg (schema | spec | uuid | location) [options] TABLE_ID
+  iceberg drop [options] (namespace | table) IDENTIFIER
+  iceberg files [options] TABLE_ID [--history]
+  iceberg rename [options] <from> <to>
+  iceberg properties [options] get (namespace | table) IDENTIFIER [PROPNAME]
+  iceberg properties [options] set (namespace | table) IDENTIFIER PROPNAME VALUE
+  iceberg properties [options] remove (namespace | table) IDENTIFIER PROPNAME
+  iceberg -h | --help | --version
+
+Arguments:
+  PARENT         Catalog parent namespace
+  IDENTIFIER     fully qualified namespace or table
+  TABLE_ID       full path to a table
+  PROPNAME       name of a property
+  VALUE          value to set
+
+Options:
+  -h --help          show this helpe messages and exit
+  --catalog TEXT     specify the catalog type [default: rest]
+  --uri TEXT         specify the catalog URI
+  --output TYPE      output type (json/text) [default: text]
+  --credential TEXT  specify credentials for the catalog`
+
+func main() {
+	args, err := docopt.ParseArgs(usage, os.Args[1:], iceberg.Version())
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	cfg := struct {
+		List     bool `docopt:"list"`
+		Describe bool `docopt:"describe"`
+		Schema   bool `docopt:"schema"`
+		Spec     bool `docopt:"spec"`
+		Uuid     bool `docopt:"uuid"`
+		Location bool `docopt:"location"`
+		Props    bool `docopt:"properties"`
+		Drop     bool `docopt:"drop"`
+		Files    bool `docopt:"files"`
+		Rename   bool `docopt:"rename"`
+
+		Get    bool `docopt:"get"`
+		Set    bool `docopt:"set"`
+		Remove bool `docopt:"remove"`
+
+		Namespace bool `docopt:"namespace"`
+		Table     bool `docopt:"table"`
+
+		RenameFrom string `docopt:"<from>"`
+		RenameTo   string `docopt:"<to>"`
+
+		Parent   string `docopt:"PARENT"`
+		Ident    string `docopt:"IDENTIFIER"`
+		TableID  string `docopt:"TABLE_ID"`
+		PropName string `docopt:"PROPNAME"`
+		Value    string `docopt:"VALUE"`
+
+		Catalog string `docopt:"--catalog"`
+		URI     string `docopt:"--uri"`
+		Output  string `docopt:"--output"`
+		History bool   `docopt:"--history"`
+		Cred    string `docopt:"--credential"`
+	}{}
+
+	if err := args.Bind(&cfg); err != nil {
+		log.Fatal(err)
+	}
+
+	var output Output
+	switch strings.ToLower(cfg.Output) {
+	case "text":
+		output = text{}
+	case "json":
+		fallthrough
+	default:
+		log.Fatal("unimplemented output type")
+	}
+
+	var cat catalog.Catalog
+	switch catalog.CatalogType(cfg.Catalog) {
+	case catalog.REST:
+		if cat, err = catalog.NewRestCatalog("rest", cfg.URI); err != nil {
```

Comment:
I don't see currently how the `--credential` is being passed to the REST catalog

Reply:
arg, i forgot to add that. I'll do so now